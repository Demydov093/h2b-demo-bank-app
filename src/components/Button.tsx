import { ReactNode } from 'react';
import { ActivityIndicator, Text, TouchableOpacity, View } from 'react-native';

type ButtonProps = {
  text: string;
  icon?: ReactNode;
  onPress?: () => void;
  height?: number;
  className?: string;
  loading?: boolean;
  disabled?: boolean;
};

export default function Button({
  text,
  icon,
  onPress,
  height,
  className = '',
  loading = false,
  disabled = false,
}: ButtonProps) {
  const isDisabled = disabled || loading;

  return (
    <TouchableOpacity
      testID="button-touchable"
      accessible
      accessibilityState={{ disabled: isDisabled }}
      onPress={onPress}
      disabled={isDisabled}
      className={`${className} flex-row items-center justify-center rounded-2xl px-4 ${
        isDisabled ? 'bg-gray-300' : 'bg-black'
      }`}
      style={{ height: height || 48 }}
    >
      {loading ? (
        <ActivityIndicator testID="activity-indicator" color="#C9FF08" size={35} />
      ) : (
        <>
          {icon && <View className="mr-2">{icon}</View>}
          <Text className="text-white text-base font-semibold">{text}</Text>
        </>
      )}
    </TouchableOpacity>
  );
}
