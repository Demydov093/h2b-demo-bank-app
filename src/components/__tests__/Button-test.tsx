import { fireEvent, render } from '@testing-library/react-native';
import { Text } from 'react-native';
import Button from '../Button';

describe('<Button />', () => {
  test('renders text correctly', () => {
    const { getByText } = render(<Button text="Click me" />);
    getByText('Click me');
  });

  test('calls onPress when pressed', () => {
    const onPressMock = jest.fn();
    const { getByTestId } = render(<Button text="Press me" onPress={onPressMock} />);
    fireEvent.press(getByTestId('button-touchable'));
    expect(onPressMock).toHaveBeenCalled();
  });

  test('renders loading indicator', () => {
    const { getByTestId, queryByText } = render(<Button text="Loading" loading />);
    getByTestId('activity-indicator');
    expect(queryByText('Loading')).toBeNull();
  });

  test('renders icon', () => {
    const { getByTestId } = render(<Button text="With Icon" icon={<Text testID="icon">I</Text>} />);
    getByTestId('icon');
  });

  test('disabled button does not call onPress', () => {
    const onPressMock = jest.fn();
    const { getByTestId } = render(<Button text="Disabled" onPress={onPressMock} disabled />);

    fireEvent.press(getByTestId('button-touchable'));
    expect(onPressMock).not.toHaveBeenCalled();
  });

  test('matches snapshot', () => {
    const tree = render(<Button text="Snapshot" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
