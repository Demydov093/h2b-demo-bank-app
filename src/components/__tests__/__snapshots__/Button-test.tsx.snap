// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 48,
      "opacity": 1,
    }
  }
  testID="button-touchable"
>
  <Text
    className="text-white text-base font-semibold"
  >
    Snapshot
  </Text>
</View>
`;
